cli/paas/install_client.go

```diff
@@ -21,88 +38,92 @@ type InstallClient struct {
+		options, err = options.Populate(kubernetes.NewInteractiveOptionsReader(os.Stdout, os.Stdin))
+		if err != nil {
+			return errors.Wrap(err, "Couldn't install carrier")
+		}
 	} else {
-		deployments.PopulateNeededOptions(kubernetes.NewDefaultOptionsReader())
+		options, err = options.Populate(kubernetes.NewDefaultOptionsReader())
+		if err != nil {
+			return errors.Wrap(err, "Couldn't install carrier")
+		}
 	}
 
-	c.showInstallConfiguration(deployments)
+	c.showInstallConfiguration(options)
 
 	// TODO (post MVP): Run a validation phase which perform
 	// additional checks on the values. For example range limits,
 	// proper syntax of the string, etc. do it as pghase, and late
 	// to report all problems at once, instead of early and
 	// piecemal.
 
-	err = c.deploySet(deployments)
+	deployment := deploymentSet["traefik"]
+	err = deployment.Deploy(c.kubeClient, c.ui, options.ForDeployment(deployment.ID()))
+	if err != nil {
+		return err
+	}
+
+	// Try to give a nip.io domain if the user didn't specify one
+	domain, err := options.GetOpt("system_domain", "")
+	if err != nil {
+		return errors.Wrap(err, "Couldn't install carrier")
+	}
+
+	err = c.fillInMissingSystemDomain(domain)
 	if err != nil {
 		return errors.Wrap(err, "Couldn't install carrier")
 	}
+	if domain.Value.(string) == "" {
+		return errors.New("You didn't provide a system_domain and we were unable to setup a nip.io domain (couldn't find and ExternalIP)")
+	}
+
+	c.ui.Success().Msg("Created system_domain: " + domain.Value.(string))
+
+	for _, deploymentName := range []string{"quarks", "gitea", "eirini", "registry", "tekton"} {
+		deployment := deploymentSet[deploymentName]
+		options := options.ForDeployment(deployment.ID())
+		err := deployment.Deploy(c.kubeClient, c.ui, options)
+		if err != nil {
+			return err
+		}
+	}
 
 	c.ui.Success().Msg("Carrier installed.")
+	c.ui.Success().Msg("Your system domain is: " + domain.Value.(string))
 
 	return nil
 }
 
 // Uninstall removes carrier from the cluster.
-func (c *InstallClient) Uninstall(cmd *cobra.Command, deploymentset *kubernetes.DeploymentSet) error {
+func (c *InstallClient) Uninstall(cmd *cobra.Command) error {
 	c.ui.Note().Msg("Carrier uninstalling...")
 
-	eiriniDeploymentID := (&deployments.Eirini{}).ID()
-	// Eirini deployment first
-	for _, d := range deploymentset.Deployments {
-		if d.ID() == eiriniDeploymentID {
-			err := d.Delete(c.kubeClient, c.ui)
-			if err != nil {
-				return err
-			}
-		}
+	deployment := deploymentSet["eirini"]
+	err := deployment.Delete(c.kubeClient, c.ui)
+	if err != nil {
+		return err
 	}
 
-	size := len(deploymentset.Deployments)
-	for index := range deploymentset.Deployments {
-		d := deploymentset.Deployments[size-index-1]
-		if d.ID() != eiriniDeploymentID {
-			err := d.Delete(c.kubeClient, c.ui)
-			if err != nil {
-				return err
-			}
+	for _, deploymentName := range []string{"tekton", "registry", "gitea", "quarks", "traefik"} {
```

Comment:
Hm. To fit my idea of using `[]deployments` instead of the map ... All the deployments are separate variables, and then two slices are made for the install and uninstall sequences outside of the special cases, i.e. treafik in install, and eirini in uninstall.

Reply:
I replied above before I saw this comment, but my comment is still valid I think (regarding the use of the same list of deployments in both cases).